### ISSUE TITLE
Error: You exceeded your current quota, please check your plan and billing details.

### ISSUE THREAD

**Issue by adamddurrant** [rubberduck-ai/rubberduck-vscode#67]
I keep getting the following error:

Error: You exceeded your current quota, please check your plan and billing details.

I have resubmitted my API key numerous times with new generation. I've also triple checked my usage and i've used next to none of my credits.

I'd really like to use this extension.

**Comment by BenjaminHoegh** [comment]
You need to have a subscription at OpenAPI before you can use there API

**Comment by adamddurrant** [comment]
So you're saying you can't use rubber duck with the free $18 API credits?
You have to be a paid subscriber?

**Comment by BenjaminHoegh** [comment]
if you still have some left, then it should work, but I do not know if OpenAI gives access to the API with the trail credits

**Comment by adamddurrant** [comment]
It does. I have barely used any of the available credits and have an API
key but still get the message. Perhaps it's an issue on the openai end 🤷‍♂️

**Comment by lgrammel** [comment]
I suspect it's an issue on the OpenAI end. Can you try to call their API directly, e.g. with:

```sh
curl https://api.openai.com/v1/completions \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer YOUR_API_KEY' \
  -d '{
  "model": "text-davinci-003",
  "prompt": "Say this is a test",
  "max_tokens": 7,
  "temperature": 0
}'
```

**Comment by womackn** [comment]
Getting the same error here as well. It seems Rubberduck's documentation is insufficient to explain what exactly is necessary for it to work aside form an API key. Guess it's back to copy/pasting code into ChatGPT.

**Comment by jamescardona11** [comment]
Check if the credits have not expired, if it doesn't work generate a new API key.
This solved my problem.

**Comment by adamddurrant** [comment]
Hi all,

I finally understood this to be a poor explanation on OpenAI's part on how free credits work and how they are visually shown in your account. 

TL;DR you can use rubberduck with free credits but the free credits for your API usage expire after 3 months. 

[This stack overflow answer](https://stackoverflow.com/questions/75898276/openai-chatgpt-gpt-3-5-api-error-429-you-exceeded-your-current-quota-please/75898717#75898717) shows you how to tell if your free credits expired.

If you want to try rubberduck with free credits after they have expired, create another openAI account or subscribe with a credit card.

**Comment by womackn** [comment]
Thanks for the response Adam, I'll take a look. Have a great week!
